src/python/roms/__init__.py

```diff
@@ -1,141 +1,101 @@
+from __future__ import annotations
+
+import base64
 import functools
-import logging
-import os
-import pathlib
-import sys
+import hashlib
+import json
+import tarfile
 import warnings
-from typing import List, Optional
-
-from ale_py import ALEInterface
-from ale_py.roms import plugins, utils
-
-# pylint: disable=g-import-not-at-top
-if sys.version_info >= (3, 9):
-    import importlib.resources as resources
-else:
-    import importlib_resources as resources
-# pylint: enable=g-import-not-at-top
-
-# Precedence is as follows:
-#  1. Internal ROMs
-#  2. External ROMs
-#  3. ROMs from atari-py.roms
-#  4. ROMs from atari-py-roms.roms
-_ROM_PLUGIN_REGISTRY: List[plugins.Plugin] = [
-    plugins.Package("ale_py.roms"),
-    plugins.EntryPoint("ale_py.roms"),
-    plugins.Package("atari_py.atari_roms"),
-    plugins.Package("atari_py_roms.atari_roms"),
-]
-
-# Environment variable for ROM discovery.
-# ale-py will search for supported ROMs in:
-#   ${ALE_PY_ROM_DIR}/*.bin
-_ROM_DIRECTORY_ENV_KEY = "ALE_PY_ROM_DIR"
-_ROM_DIRECTORY_ENV_VALUE = os.environ.get(_ROM_DIRECTORY_ENV_KEY, None)
-
-if _ROM_DIRECTORY_ENV_VALUE is not None:
-    _ROM_PLUGIN_REGISTRY.append(plugins.Directory(_ROM_DIRECTORY_ENV_VALUE))
-
-
-def _resolve_rom(name: str) -> Optional[pathlib.Path]:
-    """Resolve a ROM path from the ROM registry."""
-    for package in _ROM_PLUGIN_REGISTRY:
-        rom_id = utils.rom_name_to_id(name)
-
-        # Resolve ROM from package
-        try:
-            rom_path = package.resolve(rom_id)
-        except ModuleNotFoundError:
-            continue
-
-        # Failed to resolve ROM
-        if rom_path is None:
-            logging.debug(f"{package} did not resolve {rom_id}.")
-            continue
-
-        # ROM isn't supported
-        resolved_id = ALEInterface.isSupportedROM(rom_path)
-        if resolved_id is None:
-            warnings.warn(
-                f"{package} contains unsupported ROM: {rom_path}",
-                category=ImportWarning,
-                stacklevel=2,
+from pathlib import Path
+
+import requests
+
+
+@functools.lru_cache(maxsize=1)
+def _get_all_rom_hashes() -> dict[str, str]:
+    # this is a map of {rom.bin : md5 checksum}
+    with open(Path(__file__).parent / "md5.json") as f:
+        return json.load(f)
+
+
+def _unpack_roms() -> None:
+    """Unpacks all roms from the tar.gz file, then matches it to the expected md5 checksum."""
+    all_roms = _get_all_rom_hashes()
+
+    # load the b64 file
+    if (Path(__file__).parent / "Roms.tar.gz.b64").exists:
```

Comment:
Shouldn't this only be done if the tar hasn't been unzipped yet?

Reply:
We aren't shipping the tar.gz, we're shipping the b64, so we need to decode -> unpack -> check sums.

Reply:
Does the unzip remove the file?

Reply:
Nope, I fixed this so gonna resolve this.